django_ratelimit/core.py

```diff
@@ -227,45 +248,96 @@ def get_usage(request, group=None, fn=None, key=None, rate=None, method=ALL,
     cache_key = _make_cache_key(group, window, rate, value, method)
 
     count = None
-    try:
-        added = cache.add(cache_key, initial_value, period + EXPIRATION_FUDGE)
-    except socket.gaierror:  # for redis
-        added = False
-    if added:
-        count = initial_value
+    if is_async:
+        async def inner():
+            try:
+                # Some caches don't have an async implementation
+                if getattr(cache, 'aadd') is not None:
+                    added = await cache.aadd(cache_key, initial_value, period + EXPIRATION_FUDGE)
+                else:
+                    added = await cache.add(cache_key, initial_value, period + EXPIRATION_FUDGE)
+            except socket.gaierror:  # for redis
+                added = False
+            if added:
+                count = initial_value
+            else:
+                if increment:
+                    try:
+                        # python3-memcached will throw a ValueError if the server is
+                        # unavailable or (somehow) the key doesn't exist. redis, on the
+                        # other hand, simply returns None.
+                        count = cache.incr(cache_key)
+                    except ValueError:
+                        pass
+                else:
+                    count = await cache.aget(cache_key, initial_value)
```

Comment:
Doesn't this need an `if getattr(cache, "aget")`?

Reply:
I didn't get any issues with it, but i should add this in just in case. Good catch.